pkgs/games/armagetronad/default.nix

```diff
@@ -154,20 +156,22 @@ let
           # No passthru, end of the line.
           # https://www.youtube.com/watch?v=NOMa56y_Was
         }
-        else if (version != srcs.${latestVersionMajor}.version) then {
+        else if (version != (srcs.${latestVersionMajor} dedicatedServer).version) then {
           # Allow a "dedicated" passthru for versions other than the default.
-          dedicated = mkArmagetron (params // {
-            dedicatedServer = true;
-          });
+          dedicated = mkArmagetron fn true;
         }
-        else (lib.mapAttrs (name: value: mkArmagetron value) (lib.filterAttrs (name: value: value.version != srcs.${latestVersionMajor}.version) srcs)) // {
-          # Allow both a "dedicated" passthru and a passthru for all the options other than the latest version, which this is.
-          dedicated = mkArmagetron (params // {
-            dedicatedServer = true;
-          });
-        };
+        else
+          (
+            lib.mapAttrs (name: value: mkArmagetron value dedicatedServer)
+            (lib.filterAttrs (name: value: (value dedicatedServer).version != (srcs.${latestVersionMajor} dedicatedServer).version) srcs)
+          ) //
+          {
+            # Allow both a "dedicated" passthru and a passthru for all the options other than the latest version, which this is.
+            dedicated = mkArmagetron fn true;
+          };
 
       meta = with lib; {
+        mainProgram = pname;
```

Comment:
The main point of mainProgram is to not rely on pname and be written out explicitly.